github-pullrequest-plugin/src/main/java/org/jenkinsci/plugins/github/pullrequest/trigger/JobRunnerForCause.java

```diff
@@ -137,8 +139,22 @@ private static boolean cancelQueuedBuildByPrNumber(final int id) {
                 NUMBER.param(String.valueOf(cause.getNumber()))
         ));
         GitHubPRBadgeAction gitHubPRBadgeAction = new GitHubPRBadgeAction(cause);
-        //TODO no way to get quietPeriod, so temporary ignore it
-        return asParameterizedJobMixIn(job).scheduleBuild2(0, new CauseAction(cause), new ParametersAction(values),
+
+        ParameterizedJobMixIn parameterizedJobMixIn = asParameterizedJobMixIn(job);
+
+        int quietPeriod = 0;
+
+        // TODO: figure out a better way to do this
+        try {
+            Object mixinJob = FieldUtils.readField(parameterizedJobMixIn, "val$job", true);
+            ParameterizedJobMixIn.ParameterizedJob parameterizedJob = (ParameterizedJobMixIn.ParameterizedJob) mixinJob;
+            quietPeriod = parameterizedJob.getQuietPeriod();
+            LOGGER.debug("Got quiet period from the project: {}", quietPeriod);
```

Comment:
you can open JMX on server and connect with debugger to it. Then place breakpoint and have real-time real-world debugging :) (even with sleeping single thread instead of all)